x/feemarket/keeper/params.go

```diff
@@ -25,11 +25,12 @@ import (
 )
 
 // GetParams returns the total set of fee market parameters.
-func (k Keeper) GetParams(ctx sdk.Context) types.Params {
-	var params types.Params
+func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
 	bz := ctx.KVStore(k.storeKey).Get(types.ParamsKey)
 	if len(bz) == 0 {
 		k.ss.GetParamSetIfExists(ctx, &params)
+		// avoid nil pointers when params are not set
+		params.FillDefaults()
```

Comment:
actually it's panic from [store.parent.Get](https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/store/cachekv/store.go#L61) before get from subspace

Reply:
why parent get will have panic?

Reply:
since [store](https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/store/cachemulti/store.go#L60) is nil when query on old block

Reply:
the store won't be nil if it's registered in the binary, just the db could be empty.

Reply:
when query old version, [cacheStore](https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/store/rootmulti/store.go#L517) will be nil right?

Reply:
hmm, that makes sense.

Reply:
I guess if we have this: https://github.com/crypto-org-chain/cosmos-sdk/pull/435, we don't need this?